### DISCUSSION TITLE
Useage as observables instead of promises

### DISCUSSION THREAD

**Discussion by karunt** [capacitor-community/bluetooth-le#592] [General]
**Is your feature request related to a problem? Please describe.**
Curious if using the BLEClient methods as observables (converting them with the "from" operator) would work just as effectively as Promises. Asking because when I convert BleClient.getConnectedDevices([]) to an observable with the "from" operator (from(BleClient.getConnectedDevices([])) and pipe the result to the "tap" operator, the "tap" operator doesn't seem to execute if the getConnectedDevices method yields an empty array. That makes me wonder if any other Rxjs operators I pipe the method's results to will get executed correctly.

**Describe the solution you'd like**
Would like to see support for observables

**Describe alternatives you've considered**

**Additional context**

**Comment by pwespi** [comment]
Even tough I personally really like RxJS, providing an RxJS interface to this plugin is out of scope. The whole Capacitor API is promise based and all official plugins are promise based.

This being said, if you are using RxJS in you app, it should be possible to wrap this plugin in an RxJS layer. There is nothing special about the promise returned by `getConnectedDevices`.
